compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -16534,25 +16715,123 @@ TR::Node *nullchkSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
 //
 TR::Node *divchkSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    {
-   // Remember the original child of the divchk. If simplification of this
-   // child causes it to be replaced by another node, the divchk is no longer
-   // needed.
-   // The child of the divchk may no longer be a divide node, due to commoning.
-   // In this case the divchk is no longer needed.
+   TR::Node * child = node->getFirstChild();
+   TR::Node * origChild = child;
+   TR::ILOpCode &childOpCode = child->getOpCode();
```

Comment:
I don't think `childOpCode` needs to be a reference, and its being one makes me suspicious that it could reflect later changes to the node pointed to by `child`. There are only two uses, and they follow almost immediately, so there are no such shenanigans going on, but it might still be a bit clearer to copy the opcode

Reply:
Fixed in commit 37c05e256709f96c1913b8fcab1b45b3e3b2dca2